ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular/index.es2015';
+
+import dancing_cat from '../img/dancing_cat.gif';
+import perf from '../js/perf';
+import Graph from '../intermittent-failures/Graph';
+import RepositoryModel from '../models/repository';
+import PerfSeriesModel from '../models/perfSeries';
+import PushModel from '../models/push';
+
+import RevisionInformation from './RevisionInformation';
+
+// TODO remove $stateParams after switching to react router
+export default class CompareSubtestDistributionView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dataLoading: this.props.dataLoading,
+      originalProject: null,
+      baseDrawingData: [],
+      newDrawingData: [],
+      newProject: null,
+      commonGraphSpecs: {
+        target: '',
+        data: [],
+        width: 1000,
+        height: 275,
+        chart_type: 'bar',
+        x_accessor: 'replicate',
+        y_accessor: 'value',
+        legend: '',
+      },
+      graphOneSpecs: {},
+      graphTwoSpecs: {},
+    };
+  }
+
+  async componentDidMount() {
+    const { $stateParams } = this.props;
```

Comment:
I realize this is a WIP but a few comments: It'd be cleaner and easier to read if some of this was broken out into separate functions. The entire block from line 49 - 58 can be broken out into a `checkQueryParams` function (when we add react router, we can check `this.props.location` in componentDidMount to determine if the query params match the last location - meaning a user navigated here from the compare page and then we don't need to do another validation check). Also, if those query params are invalid - we should display an error message and not fetch the series data.

Reply:
Although I'm wondering if we even need to check them... do people ever manually add query parameters versus just navigating from the comparesubtests view?

Reply:
> Although I'm wondering if we even need to check them... do people ever manually add query parameters versus just navigating from the comparesubtests view?

I don't think they manually add query parameters.